test/src/main/java/io/strimzi/test/BaseITST.java

```diff
@@ -0,0 +1,140 @@
+    public static final KubeClient<?> KUBE_CLIENT = CLUSTER.client();
+
+    private List<String> deploymentResources = Collections.EMPTY_LIST;
+    private List<String> deploymentNamespaces = Collections.EMPTY_LIST;
+    private Map<File, String> clusterOperatorMap = Collections.EMPTY_MAP;
+
+    protected String testClass;
+    protected String testName;
+
+    private void applyClusterOperatorInstallFiles() {
+        TimeMeasuringSystem.setTestName(testClass, testClass);
+        TimeMeasuringSystem.startOperation(Operation.CO_CREATION);
+        clusterOperatorMap = Arrays.stream(new File(CO_INSTALL_DIR).listFiles()).sorted().filter(file ->
+                !file.getName().matches(".*(Binding|Deployment)-.*")
+        ).collect(Collectors.toMap(file -> file, f -> TestUtils.getContent(f, TestUtils::toYamlString), (x, y) -> x, LinkedHashMap::new));
+        for (Map.Entry<File, String> entry : clusterOperatorMap.entrySet()) {
+            LOGGER.info("Applying configuration file: {}", entry.getKey());
+            KUBE_CLIENT.clientWithAdmin().applyContent(entry.getValue());
+        }
+        TimeMeasuringSystem.stopOperation(Operation.CO_CREATION);
+    }
+
+    private void deleteClusterOperatorInstallFiles() {
+        TimeMeasuringSystem.setTestName(testClass, testClass);
+        TimeMeasuringSystem.startOperation(Operation.CO_DELETION);
+        for (Map.Entry<File, String> entry : clusterOperatorMap.entrySet()) {
+            LOGGER.info("Removing configuration file: {}", entry.getKey());
+            KUBE_CLIENT.clientWithAdmin().deleteContent(entry.getValue());
+        }
+        TimeMeasuringSystem.stopOperation(Operation.CO_DELETION);
+    }
+
+    private void createNamespaces(String useNamespace, List<String> namespaces) {
+        deploymentNamespaces = namespaces;
+        for (String namespace: namespaces) {
+            LOGGER.info("Creating namespace: {}", namespace);
+            KUBE_CLIENT.createNamespace(namespace);
+            KUBE_CLIENT.waitForResourceCreation("Namespace", namespace);
+            LOGGER.info("Namespace {} created", namespace);
+        }
+        LOGGER.info("Using namespace {}", useNamespace);
+        KUBE_CLIENT.namespace(useNamespace);
+    }
+
+    protected void createNamespaces(String useNamespace) {
+        createNamespaces(useNamespace, Collections.singletonList(useNamespace));
+    }
+
+    protected void deleteNamespaces() {
+        for (String namespace: deploymentNamespaces) {
+            LOGGER.info("Deleting namespace: {}", namespace);
+            KUBE_CLIENT.deleteNamespace(namespace);
+            KUBE_CLIENT.waitForResourceDeletion("Namespace", namespace);
+            LOGGER.info("Namespace {} deleted", namespace);
+        }
+        LOGGER.info("Using namespace {}", CLUSTER.defaultNamespace());
+        KUBE_CLIENT.namespace(CLUSTER.defaultNamespace());
+    }
+
+    protected void createCustomResources(List<String> resources) {
```

Comment:
Javadoc, include whether the given resources are literal YAML or filenames or classpath resources